mllib/src/main/scala/org/apache/spark/ml/classification/NaiveBayes.scala

```diff
@@ -129,29 +129,49 @@ class NaiveBayesModel private[ml] (
       throw new UnknownError(s"Invalid modelType: ${$(modelType)}.")
   }
 
-  override protected def predict(features: Vector): Double = {
+  override val numClasses: Int = pi.size
+
+  private def posteriorProbabilities(logProb: DenseVector) = {
```

Comment:
For this, multinomialCalculation, and bernoulliCalculation, can you just use the versions of the code from spark.mllib?  It looks like they can be made static easily by passing model parameters as needed.  That will reduce code duplication.